scipy/interpolate/_bsplines.py

```diff
@@ -606,6 +607,174 @@ def _process_deriv_spec(deriv):
         ords, vals = [], []
     return np.atleast_1d(ords, vals)
 
+def _woodbury_algorithm(A, ur, ll, b, k):
+    '''
+    Implementation of Woodbury algorithm applied to banded
+    matrices with two blocks in upper right and lower left
+    corners.
```

Comment:
"Solve a cyclic banded linear system with upper right and lower blocks of size `k` using the Woodbury formula" --- is this a fair description of this routine?

Then ASCII pseudocode structure of the matrix would help after the sentence of in the Notes section.